lib/commonio.c

```diff
@@ -157,7 +157,17 @@ static int do_lock_file (const char *file, const char *lock, bool log)
 	if (write (fd, buf, (size_t) len) != len) {
 		if (log) {
 			(void) fprintf (stderr,
-			                "%s: %s: %s\n",
+			                "%s: %s file write error: %s\n",
+			                Prog, file, strerror (errno));
+		}
+		(void) close (fd);
+		unlink (file);
+		return 0;
+	}
+	if (fsync (fd) == -1) {
```

Comment:
`fdatasync()` is better (faster), since mtime is not meaningful for this file.

Reply:
Updated with `fdatasync()`